tests/test_gen_utils.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+import pytest
+import ndmg
+from ndmg.utils.cloud_utils import s3_get_data
+from pathlib import Path
+from ndmg.utils.gen_utils import create_datadescript, DirectorySweeper
+
+
+@pytest.fixture
+def input_dir_tree(tmp_path):
+    data = {"002547": [1,2], "002548": [1], "002449": [1,2,3]}
+    for sub, session in data.items():
+        for ses in session:
+            info = f"sub-{sub}/ses-{ses}"
+            anat = os.path.join(tmp_path, info, "anat")
+            dwi = os.path.join(tmp_path, info, "dwi")
+            # make directories and files
+            os.makedirs(anat)
+            os.makedirs(dwi)
+            tmpfilepath = os.path.join(anat, f"sub-{sub}_ses-{ses}_T1w.nii.gz")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.bval")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.bvec")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+            tmpfilepath = os.path.join(dwi, f"sub-{sub}_ses-{ses}_dwi.nii.gz")
+            with open(tmpfilepath, "x") as f:
+                f.write("placeholder text")
+
+    # Empty Directory
+    os.makedirs(os.path.join(tmp_path, "sub-002"))
+    # Create non-BIDS files
+    tmpfilepath = os.path.join(anat, f"dummy.txt")
+    with open(tmpfilepath, "x") as f:
+        f.write("placeholder text")
+    tmpfilepath = os.path.join(dwi, f"sub-55.nii.gz")
+    with open(tmpfilepath, "x") as f:
+        f.write("placeholder text")
+
+    # Create json file of input data
+    create_datadescript(os.path.join(tmp_path))
+
+    return tmp_path
+
+
+def test_DirectorySweeper(input_dir_tree):
+    data = {"002547": [1,2], "002548": [1], "002449": [1,2,3]}
```

Comment:
in general, whenever the same variable is defined twice across two functions, I think we should aim to just define it once unless there's a good reason not to
Here, we can probably stick `data` into the outer (file) namespace, since we'll likely be moving this stuff into a different file module